devicetypes/stelpro/stelpro-ki-thermostat.src/stelpro-ki-thermostat.groovy

```diff
@@ -0,0 +1,558 @@
+				attributeState("VALUE_DOWN", action: "decreaseHeatSetpoint")
+			}
+			tileAttribute("device.thermostatOperatingState", key: "OPERATING_STATE") {
+				attributeState("idle", backgroundColor:"#44b621")
+				attributeState("heating", backgroundColor:"#ffa81e")
+			}
+			tileAttribute("device.thermostatMode", key: "THERMOSTAT_MODE") {
+				attributeState("heat", label:'${name}')
+				attributeState("eco", label:'${name}')
+			}
+			tileAttribute("device.heatingSetpoint", key: "HEATING_SETPOINT") {
+				attributeState("heatingSetpoint", label:'${currentValue}')
+			}
+		}
+		standardTile("mode", "device.thermostatMode", width: 2, height: 2) {
+			state "heat", label:'${name}', action:"switchMode", nextState:"eco", icon:"http://cdn.device-icons.smartthings.com/Home/home29-icn@2x.png"
+			state "eco", label:'${name}', action:"switchMode", nextState:"heat", icon:"http://cdn.device-icons.smartthings.com/Outdoor/outdoor3-icn@2x.png"
+		}
+		valueTile("heatingSetpoint", "device.heatingSetpoint", width: 2, height: 2) {
+			state "temperature", label:'Setpoint ${currentValue}', backgroundColors:[
+					[value: 31, color: "#153591"],
+					[value: 44, color: "#1e9cbb"],
+					[value: 59, color: "#90d2a7"],
+					[value: 74, color: "#44b621"],
+					[value: 84, color: "#f1d801"],
+					[value: 95, color: "#d04e00"],
+					[value: 96, color: "#bc2323"]
+			]
+			state "--", label:'--', backgroundColor:"#bdbdbd"
+		}
+		standardTile("refresh", "device.refresh", decoration: "flat", width: 2, height: 2) {
+			state "default", action:"refresh.refresh", icon:"st.secondary.refresh"
+		}
+		
+		main ("thermostatMulti")
+		details(["thermostatMulti", "mode", "heatingSetpoint", "refresh"])
+	}
+}
+
+def getSupportedThermostatModes() {
+	modes()
+}
+
+def getThermostatSetpointRange() {
+	if (getTemperatureScale() == "C") {
+		[5, 30]
+	}
+	else {
+		[41, 86]
+	}
+}
+
+def getHeatingSetpointRange() {
+	thermostatSetpointRange
+}
+
+def setupHealthCheck() {
+	// Device-Watch simply pings if no device events received for 32min(checkInterval)
+	sendEvent(name: "checkInterval", value: 2 * 15 * 60 + 2 * 60, displayed: false, data: [protocol: "zwave", hubHardwareId: device.hub.hardwareID])
+}
+
+def installed() {
+	setupHealthCheck()
+
+	sendEvent(name: "supportedThermostatModes", value: supportedThermostatModes, displayed: false)
+	sendEvent(name: "thermostatSetpointRange", value: thermostatSetpointRange, displayed: false)
+	sendEvent(name: "heatingSetpointRange", value: heatingSetpointRange, displayed: false)
+}
+
+def updated() {
+	setupHealthCheck()
+
+	sendEvent(name: "supportedThermostatModes", value: supportedThermostatModes, displayed: false)
+	sendEvent(name: "thermostatSetpointRange", value: thermostatSetpointRange, displayed: false)
+	sendEvent(name: "heatingSetpointRange", value: heatingSetpointRange, displayed: false)
+}
+
+def parse(String description) {
+	return parseCalled(description)
+}
+
+def parseCalled(String description) {
```

Comment:
Seems like this extra function can be removed and just put this all in `parse`. Also remove the commented out code if it's not needed.